### ISSUE TITLE
Startus page https certificate expiration

### ISSUE THREAD

**Issue by plegrand1** [louislam/uptime-kuma#5313]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

Hello,
I'm having a small problem checking the expiry of an https certificate.
This one has a particularity in that it is self-signed.
On Kuma's main page, it appears with the number of days remaining.
![image](https://github.com/user-attachments/assets/feb365a2-2117-4a1a-a8e4-cc2c4c6d8897)
However, on the status page, the number of days remaining does not appear.
![image](https://github.com/user-attachments/assets/564880a3-04b7-4162-a101-3969dd4ef58b)
Is this a bug or a misconfiguration on my part?
Thank you for your help.
Best regards,

### 🐻 Uptime-Kuma Version

1.23.15

### 💻 Operating System and Arch

Debian

### 🌐 Browser

Firefox

### 🖥️ Deployment Environment

- Runtime: 
- Database: sqlite
- Filesystem used to store the database on: 
- number of monitors:

**Comment by apio-sys** [comment]
For the expiration date to appear on the status page, you need to switch "Show Certificate Expiry" to on on the bottom left when editing the status page.
![Screenshot from 2024-11-05 16-45-16](https://github.com/user-attachments/assets/b3a6119c-71bc-43e9-ae43-0fe2c3f5782a)

**Comment by apio-sys** [comment]
It works for me but the expiry notice shows below the monitor with the wording "Expiration Cert SSL: nn jours" if running in French. Yours shows "Certificats HTTPS". Looks like we are not running the same version? (I have 1.23.15 also). But haven't tested with self-signed. Let me try that in a bit.

**Comment by apio-sys** [comment]
Indeed, I could reproduce, with a self-signed cert the remaining days do not show anymore in the Status Page. We should either fix it or document that such certs will not show on status pages (since this is more a "customer facing" feature we might not want to show self-signed certs here?). Could find a why in the documentation. I guess this comes from here `/src/components/PublicGroupList.vue` but I haven't a dev environment running right now to test further.

**Comment by plegrand1** [comment]
Hello,
is there a solution to correct this problem?
I understand that it's just a display problem, but I'd be happy if all my certificate expirations (signed or self-signed) could be displayed on Status page.
Thanks for your help

**Comment by Tragen** [comment]
For most self-signed certificates, I've installed my CA files where Uptime Kuma is running, but there are a lot more of internal devices which create self-signed certificates automatically, and I want to monitor them.

**Comment by Tragen** [comment]
Still important

**Comment by Tragen** [comment]
Still important

**Comment by cc13com** [comment]
I'm also interested in this feature since a lot of my servers running with letsencrypt certificates.

**Comment by apio-sys** [comment]
For self-signed certs, it affects status pages. Those are typically customer-facing. That was the point of this ticket. If you'd like to see other changes, I'd suggest to create a new proposition.

> I'm also interested in this feature since a lot of my servers running with letsencrypt certificates.

@cc13com This already works as expected and documented and is not related to this ticket.

**Comment by Tragen** [comment]
It's open because it's not fixed. Still important.
This is so lazy to just close issues.

**Comment by CommanderStorm** [comment]
Since what you’re describing sounds more like a feature request than a help ticket, I’ve re-labeled it as such. Feel free to look into why self-signed certificates behave different.

I would be open to reviewing an PR, but won't myself invest time to develop an fix since I don't use self-signed certificates
